### ISSUE TITLE
Version change in next branch fails setup.py version check

### ISSUE THREAD

**Issue by m0n0cle** [nccgroup/depthcharge#72]
**Describe the bug**

The "-dev" added to the version number in `next` branch `python/depthcharge/version.py` causes the version check in `python/setup.py` to fail.

**Depthcharge Version**

https://github.com/nccgroup/depthcharge/commit/31042d53992706dbcf3c4e901a94d2dbe93305c9

**Target System**

Attempted install on Ubuntu 20.04
Python version: 3.8.5

**Additional context**

Suggested change to regex in `python/setup.py`:
`r"__version__\s*=\s*'(?P<version>[0-9]+\.[0-9]+\.[0-9]+(-[a-zA-Z0-9]+)?)'"`

**Comment by jynik** [comment]
Thanks for reporting this @m0n0cle - I ran into this myself while working on the v0.3.x series and failed to address it. Yikes - my apologies.

With regard to `-dev` -- I learned that this actually violates the versioning scheme mandated by PEP440.  It should be `.dev0` or something to that effect.  I'll need to remember this moving forward, but will still make the version regex more flexible.  

Nonetheless, I shouldn't have a `next` commit broken like this -- need to set up a proper CI pipeline.  Created #73 to track  this.

Additionally, the fix for #23 will require an API addition, so I'll need to bump this to `0.3.0.dev0`
